main.cpp

```diff
@@ -210,285 +239,379 @@ struct Museum
+            mainOutputVoltage = mainsInputVoltage * powerTransformerMainRatio;
+        else
+            mainOutputVoltage = 0.f;
+    }
+}
+
+void PowerSupply::blowFuse()
+{
+    fuseState = false;
+    mainOutputVoltage = 0;
+    heaterOutputVoltage = 0;
+}
+
 struct OutputSection
 {
-    // 1) the number of output tubes (int)
     int numberOfOutputTubes = 2;
-    // 2) the grid bias voltage (float)
     float gridBiasVoltage = -50.f;
-    // 3) the output transformers primary impedance in Ohms (double)
     double outputTransformerPrimaryImpedance { 2600 };
-    // 4) the output transformers secondary impedance in Ohms (float)
     float outputTransformerSecondaryImpedance = 8.f;
-    // 5) the maximum output power in watts (float)
     float maxOutputPower = 50.f;
 
-    // 1) Warm up tubes
     void warmUpTubes(float timeInMSTillWarm);
-    // 2) Admust master output volume
     void adjustMasterOutputVolume(float newVolume);
-    // 3) Amplify line level audio to a speaker level signal
     float amplifyLineLevelAudioToSpeakerLevel(float inputSignal);
+
+private:
+    bool _tubesAreWarm;
+    float _masterOutputGain;
+    float _gainFactor = numberOfOutputTubes * 24.f;
+
 };
-/*
-Thing 7) Preamp Section
-5 properties:
-    1) the number of preamp tubes (int)
-    2) the gain setting for channel 1 (float)
-    3) the gain setting for channel 2 (float)
-    4) the active channel (int)
-    5) the cathode bias resistor value for the second gain stage (int)
-3 things it can do:
-    1) warm up tubes
-    2) Adjust channel gain setting
-    3) Amplify guitar level signal to line voltage level
- */
+
+void OutputSection::warmUpTubes(float timeInMSTillWarm)
+{
+    if (timeInMSTillWarm < 1000)
+    {
+        _tubesAreWarm = true;
+    }
+}
+
+void OutputSection::adjustMasterOutputVolume(float newVolume)
+{
+    _masterOutputGain = newVolume;
+}
+
+float OutputSection::amplifyLineLevelAudioToSpeakerLevel(float inputSignal)
+{
+    if(_tubesAreWarm)
+        return (inputSignal * _gainFactor * _masterOutputGain);
+    return 0.f;
+}
+
 struct PreampSection
 {
-    // 1) the number of preamp tubes (int)
     int numberOfPreampTubes = 2;
-    // 2) the gain setting for channel 1 (float)
     float channelOneGain = 0.33f;
-    // 3) the gain setting for channel 2 (float)
     float channelTwoGain = 0.33f;
-    // 4) the active channel (int)
     int activeChannel = 0;
-    // 5) the cathode bias resistor value for the second gain stage (int)
     int stageTwoBiasResistorValue = 820;
 
-    // 1) warm up tubes
     void warmUpTubes(float timeInMSTillWarm);
-    // 2) Adjust channel gain setting
     void adjustChannelGain(int channel, float gain);
-    // 3) Amplify guitar level signal to line voltage level
-    float amplifyGuitarSignalToSpeakerLevel(float inputSignal);
+    float amplifyGuitarSignalToLineLevel(float inputSignal);
+
+private:
+    bool _tubesAreWarm;
+    float _gainFactor = numberOfPreampTubes * 50.f;
 };
```

Comment:
don't use `private`.  private is covered in Project 7
Don't use underscores in member variable names
- [ ] revise